u3-views/features/videoController.js

```diff
@@ -0,0 +1,19 @@
+const { getVideo, getSuggestions } = require('../_services/fakeapi')
+
+const videoPlayer = (req, res) => {
+  const videoId = req.params.id
+  const video = getVideo(videoId)
+  const teases = getSuggestions(6)
+  res.render('video-player', { video: video, teases: teases })
+}
+const videoHome = (req, res) => {
+  const videoId = req.params.id
+  const video = getVideo(videoId)
```

Comment:
เส้นทางนี้คือ `/videos`. ไม่มีพารามิเตอร์ `id`. `videoId` & `video` จะไม่ถูกกำหนดเสมอ และคุณไม่ต้องการมันในมุมมองของคุณ (ฉันชอบเรียกตัวแปรข้อมูลว่า `cards` เพื่อหลีกเลี่ยงความสับสนกับ 'video-tease.hbs`

This route is `/videos`. There is no `id` parameter. `videoId` & `video` will always be undefined. And you don't need them in your view. I also prefer to call the data variable `cards`, to avoid confusion with `video-tease.hbs`.

```js
const videoHome = (req, res) => {
  const cards = getSuggestions(30)
  res.render('video-home', { cards: cards })
}
```